F telios 0624

suleymancelikel opened the pull request.

suleymancelikel committed:
[Telios-0624]: global state and getMessageById action added

suleymancelikel committed:
[Telios-0624]: filteredInboxSelector added

suleymancelikel committed:
[Telios-0624]: refactor for saveMailToDB method in fetcheData

suleymancelikel committed:
[Telios-0624]: Filter Option edited and added functionality

vitalikgriza commented on src/components/MailList.tsx:
```diff
@@ -173,15 +171,50 @@ export const MailList = ({
   </View>
 );
+
+const styles = StyleSheet.create({
```
Could you please move styles to separate file, if needed create a folder for component

vitalikgriza commented on src/store/global.ts:
```diff
@@ -0,0 +1,36 @@
+  name: 'global',
+  initialState,
+  reducers: {
+    setMessageListFilters(state, action: PayloadAction<MessageListFilters>) {
```
could you please rename action like "update" or something, it's ease to confuse with setState function

vitalikgriza commented on src/screens/InboxScreen.tsx:
```diff
@@ -44,12 +45,17 @@ export const InboxScreen = (props: InboxScreenProps) => {
   };
 
   const onSelectEmail = (emailId: string) => {
+    // dispatch(getMessageById({ id: emailId }));
```
Remove unnecessary comments

vitalikgriza reviewed (changes requested):
Please merge all commits into one and omit "[Telios-0624]" in commit message

suleymancelikel committed:
The requested changes have been done

suleymancelikel committed:
[Telios-0624]: global state and getMessageById action added

[Telios-0624]: filteredInboxSelector added

[Telios-0624]: refactor for saveMailToDB method in fetcheData

[Telios-0624]: Filter Option edited and added functionality

The requested changes have been done

suleymancelikel committed:
Merge branch 'F_Telios-0624' of https://github.com/Telios-org/rn-mobile-client.git; branch 'F_Telios-0624' of https://github.com/Telios-org/rn-mobile-client into F_Telios-0624
